### PR TITLE
Postgres

### PR DESCRIPTION
**Pull request by miraclestyle**
Tickets:
- [Install Postgres](https://trello.com/c/iZnSDQ05/14-install-postgres).
- [Setup backend APIs.](https://trello.com/c/P2MA7qCl/16-setup-backend-apis).
- [Extend the API to support all four requests](https://trello.com/c/6lsENA3M/17-extend-the-api-to-support-all-four-requests).
- [Fine-tune data import in Postgres](https://trello.com/c/nIho00bn/22-fine-tune-data-import-in-postgres).
- [Fine-tune data generator](https://trello.com/c/IowIWNoc/21-fine-tune-data-generator).

### CODE REVIEW

**Review by matt-wigg — APPROVED** [review]
Solid - can't fault anything. Great variable naming - great layout - great separations. 🚀

File: server/database/generate/utils.js [hunk 1]
```diff
@@ -136,11 +136,85 @@ const checkOutHelper = () => {
...
+// };
+
+// ***************LESS DATA***************
+
 const getRecord = () => {
   const adjOne = faker.random.arrayElement(constants.adj);
   const adjTwo = faker.random.arrayElement(constants.adj);
   const city = faker.address.city();
-  const name = `${adjOne} ${adjTwo} ${city}`;
+  const hostelName = `${adjOne} ${adjTwo} ${city}`;
```
Comment (matt-wigg) [inline server/database/generate/utils.js:217]: Nice work on making your variable names more specific!

File: server/router/index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const db = require('../database');
+
+const getRecordById = (req, res, next) => {
+  db.postgresql.getRecordById(req.params.id)
+    .then((data) => {
+      // console.log('getRecord data:', data);
+      res.status(200);
+      res.json(data.rows);
```
Comment (matt-wigg) [inline server/router/index.js:8]: Not a big deal, but you could maybe streamline this a little with `res.status(200).json(data.rows)`

File: server/database/postgresql/schema.sql [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+--   descriptions_id SMALLINT NOT NULL,
+--   rules_id SMALLINT NOT NULL,
+--   addresses_id SMALLINT NOT NULL,
+--   PRIMARY KEY (id)
+-- );
+
+-- ALTER TABLE full_listing ADD FOREIGN KEY (name_id) REFERENCES hostels (id);
+-- ALTER TABLE full_listing ADD FOREIGN KEY (descriptions_id) REFERENCES descriptions (id);
+-- ALTER TABLE full_listing ADD FOREIGN KEY (rules_id) REFERENCES rules (id);
+-- ALTER TABLE full_listing ADD FOREIGN KEY (addresses_id) REFERENCES addresses (id);
```
Comment (matt-wigg) [inline server/database/postgresql/schema.sql:125]: Are you keeping all this in comments for reference?

File: server/database/postgresql/queries.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const db = require('./client.js');
+
+const getRecordById = (id) => {
+  const q = 'SELECT * FROM hostels WHERE id = $1';
```
Comment (matt-wigg) [inline server/database/postgresql/queries.js:4]: You use template literals in some of your other requests. Could you do the same here? 😃

**Review by matt-wigg — APPROVED** [review]
Nice job! Just a few tiny things to consider, but overall this is awesome - great work! 🚀

File: server/database/postgresql/schema.sql (hunk 3 above)
Comment (miraclestyle) [reply server/database/postgresql/schema.sql:125]: Yes. I find them handy during the tunning. Once I am done with it, I will remove them.

File: server/database/postgresql/queries.js (hunk 4 above)
Comment (miraclestyle) [reply server/database/postgresql/queries.js:4]: The Airbnb style linter throws junk at me if I try. :D

File: server/router/index.js (hunk 2 above)
Comment (miraclestyle) [reply server/router/index.js:8]: Thanks. Surely will do.

**Pull request merged**